### PR TITLE
ENH Set Published button dirty state when Link is an unpublished state

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-linkfield/issues/152

I've tested this for:
- Links on Pages
- Links on Elemental
- Links on DataObjects managed by ModelAdmins

It will set the "Publish" button to a dirty state when the LinkPickerTitle has either a 'Draft' or a 'Modified' versionedState.  It won't change the state of "Save" button as LinkField constantly saves everything via AJAX.

Note that for Elemental this is only true if the Link if modifed via the link modal.  The "Save" button will still get a dirty state when new links are added or deleted.  This is due to existing logic in Elemental to that will trigger the jQuery changetracker which will alter both the Save and the Publish buttons.  There's nothing we can really do about this without a lot of hacking around it, and I don't think this is worth it since a dirty Save button is far from the worse thing ever.

I've also created a new issue for the all the CMS [inconsistency](https://github.com/silverstripe/silverstripe-admin/issues/1665) I discovered about the changetracker while doing this work

### CODE REVIEW

File: client/src/components/LinkField/LinkField.js [hunk 1]
```diff
@@ -136,7 +137,7 @@ const LinkField = ({
   /**
    * Update the component when the 'Delete' button in the LinkPicker is clicked
    */
-  const onDelete = (linkID, deleteType) => {
+  const handleDelete = (linkID, deleteType) => {
```
Comment (emteknetnz) [inline client/src/components/LinkField/LinkField.js:140]: This is unrelated refactoring to follow the consistent naming of defining 'handle*' methods and passing them to 'on*' props

File: client/src/components/LinkField/LinkField.js [hunk 2]
```diff
@@ -172,6 +173,42 @@ const LinkField = ({
...
+      '.cms-edit-form button[data-text-alternate]#Form_ItemEditForm_action_doPublish'
+    ].join(',');
+    const publishButton = document.querySelector(cssSelector);
+    if (!publishButton) {
+      return;
+    }
+    const dataBtnAlternateRemove = publishButton.getAttribute('data-btn-alternate-remove') || '';
+    dataBtnAlternateRemove.split(' ').forEach((className) => {
+      publishButton.classList.remove(className);
+    });
```
Comment (sabina-talipova) [inline client/src/components/LinkField/LinkField.js]: Everything works perfect, as expected. But there is a small issue here. If somehow we pass empty string as class name then we get `Uncaught DOMException: Failed to execute 'remove' on 'DOMTokenList': The token provided must not be empty.`. Probably we should check that `dataBtnAlternateRemove`, `dataBtnAlternateAdd `, `dataTextAlternate ` are not empty string before.

**Review by sabina-talipova — CHANGES_REQUESTED** [review]

File: client/src/components/LinkField/LinkField.js (hunk 2 above)
Comment (emteknetnz) [reply client/src/components/LinkField/LinkField.js]: Have added a couple of `if (className) {` checks

**Review by sabina-talipova — APPROVED** [review]
LGTM. Tested locally.

**Pull request merged**